itest/lnd_open_channel_test.go

```diff
@@ -768,3 +768,80 @@ func testFundingExpiryBlocksOnPending(ht *lntest.HarnessTest) {
 	chanPoint := lntest.ChanPointFromPendingUpdate(update)
 	ht.CloseChannel(alice, chanPoint)
 }
+
+// testSimpleTaprootChannelActivation ensures that a simple taproot channel is
+// active if the initiator stays disconnected in between channel opening and
+// channel confirmation but reconnects after the channel is confirmed.
+func testSimpleTaprootChannelActivation(ht *lntest.HarnessTest) {
+	// Grab the channel participants.
+	alice, bob := ht.Alice, ht.Bob
+
+	// Restart Alice and Bob to activate simple taproot channel feature.
+	simpleTaprootChanArgs := lntest.NodeArgsForCommitType(
+		lnrpc.CommitmentType_SIMPLE_TAPROOT,
+	)
+	ht.RestartNodeWithExtraArgs(alice, simpleTaprootChanArgs)
+	ht.RestartNodeWithExtraArgs(bob, simpleTaprootChanArgs)
+
+	// Make sure Alice and Bob are connected.
+	ht.EnsureConnected(alice, bob)
+
+	// Create simple taproot channel opening parameters.
+	params := lntest.OpenChannelParams{
+		FundMax:        true,
+		CommitmentType: lnrpc.CommitmentType_SIMPLE_TAPROOT,
+		Private:        true,
+	}
+
+	// Alice opens the channel to Bob.
+	pendingChan := ht.OpenChannelAssertPending(alice, bob, params)
+
+	// We'll create the channel point to be able to close the channel once
+	// our test is done.
+	chanPoint := &lnrpc.ChannelPoint{
+		FundingTxid: &lnrpc.ChannelPoint_FundingTxidBytes{
+			FundingTxidBytes: pendingChan.Txid,
+		},
+		OutputIndex: pendingChan.OutputIndex,
+	}
+
+	// We disconnect Bob from Alice before the channel is confirmed
+	// on-chain. Our expectation is that once the channel is confirmed and
+	// Alice reconnects, the channel will show as active on Alice's side.
+	alice.RPC.DisconnectPeer(bob.PubKeyStr)
+
```

Comment:
so if we connect the nodes again here, we'd see this error in Alice's log,
```
2023-10-10 19:31:50.648 [WRN] HSWC: ChannelLink(66bea0e467d87ade4430bcf08d485dbdfaa40f092bb88337c057c34b1df4007c:0): error when syncing channel states: unable to init remote nonce: pending verification nonce is not set
2023-10-10 19:31:50.648 [ERR] HSWC: ChannelLink(66bea0e467d87ade4430bcf08d485dbdfaa40f092bb88337c057c34b1df4007c:0): failing link: unable to synchronize channel states: unable to init remote nonce: pending verification nonce is not set with error: unable to resume channel, recovery required
```

I think it indicates there's a connection attempt in your regtest environment - since this is a permanent connection, alice(zane) will try to reconnect to bob.

Reply:
nice catch, I've now identified the root cause here

Reply:
Thanks for the hint @yyforyongyu. @Crypt-iQ, will complete this test in a bit.